tests/cbmc/proofs/s2n_constant_time_pkcs1_unpad_or_dont/s2n_constant_time_pkcs1_unpad_or_dont_harness.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "utils/s2n_safety.h"
+
+#include <sys/param.h>
+#include <assert.h>
+
+#include <cbmc_proof/proof_allocators.h>
+
+void s2n_constant_time_pkcs1_unpad_or_dont_harness() {
+    /* Non-deterministic inputs. */
+    const uint32_t len;
+    const uint32_t destlen;
+    const uint32_t srclen;
+    const uint8_t dont;
+    __CPROVER_assume(len < MAX_ARR_LEN);
+    __CPROVER_assume(destlen >= len);
+    __CPROVER_assume(srclen >= len + 1 && srclen <= (MAX_ARR_LEN + 3));
```

Comment:
Add a comment why do we need this assumption.

Reply:
Where the + 1 comes from? You could also replace `(MAX_ARR_LEN + 3)` with `BOUND`